cloud/disk_manager/test/acceptance/test_runner/base_acceptance_test_runner.py

```diff
@@ -322,12 +322,13 @@ def _find_or_create_eternal_disk(self, disk_name_prefix) -> Ycp.Disk:
         return disk
 
     @contextlib.contextmanager
-    def instance_policy_obtained(self, handler):
+    def instance_policy_obtained(self, handler, keep_instance_on_error=False):
         ctx = self._instance_policy.obtain()
         try:
             instance = ctx.__enter__()
         except Exception as e:
             handler(e)
+            ctx._auto_delete = not keep_instance_on_error
```

Comment:
Why do we need this? We do not delete not created  vm anyway

Reply:
We delete it here
https://github.com/ydb-platform/nbs/blob/5f4f11a38f5a9c3f60da5c2aeb2d71a3b71e71a9/cloud/disk_manager/test/acceptance/test_runner/lib/instance_policy.py#L69

We create instance using context manager https://github.com/ydb-platform/nbs/blob/5f4f11a38f5a9c3f60da5c2aeb2d71a3b71e71a9/cloud/disk_manager/test/acceptance/test_runner/sync_acceptance_test_runner.py#L79 and after everything is finished - we remove it. This flag is also true if we add --debug flag to command line

Reply:
Once again, you set ctx._auto_delete when an exception occured in the __enter__  part. 
https://github.com/ydb-platform/nbs/pull/1423/files#diff-b4f86b51bae84e70cab38346d5a3fc0c82f4f5f9c0b53a9cde3f620aefd24509L60
After that, the exception gets raised further and ctx.__exit__ is not called.